lib/mixlib/install/backend/artifactory.rb

```diff
@@ -54,24 +56,31 @@ def info
           end
         end
 
-        private
+        def product_versions
+          params = {
+            g: "com.getchef",
+            a: options.product_name,
+            repos: ARTIFACTORY_REPOSITORY
+          }
+          results = client.get("/api/search/versions", params)["results"]
```

Comment:
I'm really worried about using any GAVC search logic in Artifactory. I would prefer we return a list of all known versions for a project (in a particular channel) using AQL (query with `omnibus.project` and return only unique `omnibus.version`). After you have that list it's possible to parse/sort everything with mixlib-versioning.

We use a similar query in chef-web-downloads:
https://github.com/chef/chef-web-downloads/blob/master/lib/chef/web/downloads/package_list.rb#L35-L43